UPGRADE.md

```diff
@@ -0,0 +1,16 @@
+# Upgrade Guide
```

Comment:
Great that you thought about adding this file!

But to be consistent with `windows` and `java` sdk let's put this file in `/docs/upgrades` and name it `2.x.x to 3.x.x.md`.

Please also add Upgrades section in readme.md like java did (https://github.com/box/box-java-sdk#upgrades).

When it comes to content of this file, try to use the structure of java/win upgrade file